app/frontend/private/pos/item-management/create.js

```diff
@@ -50,3 +54,42 @@ create.addEventListener("click", () => {
             console.log(error);
         });
 });
+
+upload.addEventListener("click", () => {
+    file.append("file", fileInput.files[0]);
+
+    fetch("/api/item/upload", {
+        method: "POST",
+        body: file,
+    })
+    .then(async (response) => {
+        const result = await response.json();
+        console.log(result.data);
+
+
+        // TODO
+        const item = {
+            category: result[0].category,
+            name: result[0].name,
+            description: result[0].description,
+            price: result[0].price,
+        };
+    
+        fetch("/api/item/add", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify(item),
+        })
+            .then((response) => {
+                response.ok ? alert("Success") : alert("Failure");
+            })
+            .catch((error) => {
+                console.log(error);
+            });
```

Comment:
I'm pretty sure this can be done inside the backend instead of making an HTTP request for each item

Reply:
Yeah I was planning on doing it directly through api.js and I'm going to remove all of this. Is it possible to call a route from within a route? i.e calling /api/item/add from within /api/item/upload?

Reply:
you don't need to call another route, just make a loop and insert the items into the database